packages/wrangler/src/deploy/deploy.ts

```diff
@@ -635,17 +651,37 @@ See https://developers.cloudflare.com/workers/platform/compatibility-dates for m
 			config.placement?.mode === "smart" ? { mode: "smart" } : undefined;
 
 		const entryPointName = path.basename(resolvedEntryPointPath);
+		const main: CfModule = {
+			name: entryPointName,
+			filePath: resolvedEntryPointPath,
+			content: content,
+			type: bundleType,
+		};
+		let sourceMaps: CfWorkerSourceMap[] | undefined = undefined;
+		if (props.sourceMaps ?? config.source_maps) {
+			const { sourceMapPath, sourceMapMetadata } =
+				bundle as Partial<BundleResult>;
```

Comment:
Can we just do the destructuring above?

Reply:
This ended up getting refactored into a separate function